File: uniffi/src/bindings/swift/templates/TopLevelFunctionTemplate.swift
{%- match func.return_type() -%}
{%- when Some with (return_type) %}

public func {{ func.name()|fn_name_swift }}({%- call swift::arg_list_decl(func) -%}) {% call swift::throws(func) %} -> {{ return_type|decl_swift }} {
    let _retval = {% call swift::to_rs_call(func) %}
    return {%- call swift::try(func) %} {{ "_retval"|lift_swift(return_type) }}
}

{% when None -%}

public func {{ func.name()|fn_name_swift }}({% call swift::arg_list_decl(func) %}) {% call swift::throws(func) %} {
    {% call swift::to_rs_call(func) %}
}
{% endmatch %}

File: uniffi_bindgen_kotlin/src/templates/TopLevelFunctionTemplate.kt
{% import "macros.kt" as kt %}
{%- let func = self.inner() %}
{%- match func.throws() -%}
{%- when Some with (throwable) %}
@Throws({{ throwable|exception_name }}::class)
{%- else -%}
{%- endmatch %}
{%- match func.return_type() -%}
{%- when Some with (return_type) %}

fun {{ func.name()|fn_name }}({%- call kt::arg_list_decl(func) -%}): {{ return_type|type_name }} {
    val _retval = {% call kt::to_ffi_call(func) %}
    return {{ "_retval"|lift_var(return_type) }}
}

{% when None -%}

fun {{ func.name()|fn_name }}({% call kt::arg_list_decl(func) %}) =
    {% call kt::to_ffi_call(func) %}
{% endmatch %}
